src/lib/UXDController.ts

```diff
@@ -28,158 +31,181 @@ export class UXDController {
+    signer: Signer
+  ): Promise<ContractTransaction> {
+    return await this.controllerContract
+      .connect(signer)
+      .redeemEth(this.market, uxdAmount, slippageAmount);
   }
 
   // ???Rework returned infos???
   public async getCollateralInfo(): Promise<UXDControllerNamespace.CollateralInfoStructOutput[]> {
     return this.controllerContract.getCollateralInfo();
   }
 
-  public approveUXD({
+  public async approveUXD({
     spender,
     amount,
     signer,
   }: {
     spender: string;
-    amount: BigNumber;
+    amount: number;
     signer: Signer;
   }): Promise<ContractTransaction> {
-    return this.uxdContract.connect(signer).approve(spender, amount);
+    const uxdAmount = ethers.utils.parseEther(amount.toString());
+    return await this.uxdContract.connect(signer).approve(spender, uxdAmount);
   }
 
-  public approveToken({
+  public async approveToken({
     contractAddress,
     spender,
     amount,
     signer,
   }: {
     contractAddress: string;
     spender: string;
-    amount: BigNumber;
+    amount: number;
     signer: Signer;
   }): Promise<ContractTransaction> {
-    return ERC20__factory.connect(contractAddress, signer).connect(signer).approve(spender, amount);
+    const ethAmount = ethers.utils.parseEther(amount.toString());
+    return await ERC20__factory.connect(contractAddress, signer).connect(signer).approve(spender, ethAmount);
   }
 
-  public allowance({
+  public async allowance({
     contractAddress,
     account,
     spender,
   }: {
     contractAddress: string;
     account: string;
     spender: string;
-  }): Promise<BigNumber> {
-    return ERC20__factory.connect(contractAddress, this.provider).allowance(account, spender);
+  }): Promise<number> {
+    const allowance = await ERC20__factory.connect(contractAddress, this.provider).allowance(account, spender);
+    return Number(ethers.utils.formatEther(allowance));
   }
 
-  public tokenBalance({
+  public async tokenBalance({
     contractAddress,
     account,
   }: {
     contractAddress: string;
     account: string;
-  }): Promise<BigNumber> {
-    return ERC20__factory.connect(contractAddress, this.provider).balanceOf(account);
+  }): Promise<number> {
+    const balance = await ERC20__factory.connect(contractAddress, this.provider).balanceOf(account);
+    return Number(ethers.utils.formatEther(balance));
   }
 
-  public uxdTotalSupply(): Promise<BigNumber> {
-    return this.uxdContract.totalSupply();
+  public async uxdTotalSupply(): Promise<number> {
+    const totalSupply = await this.uxdContract.totalSupply();
+    return Number(ethers.utils.formatEther(totalSupply));
   }
 
-  public mintedPerCollateral(token: string): Promise<BigNumber> {
-    return this.controllerContract.mintedPerCollateral(token);
+  public async mintedPerCollateral(token: string): Promise<number> {
+    const minted = await this.controllerContract.mintedPerCollateral(token);
+    return Number(ethers.utils.formatEther(minted));;
```

Comment:
Maybe rename `formatEther` to include the idea of, Native format => UI format

Reply:
`formatEther` is an ethers.js library function, I don't have control over the name

Reply:
oh yeah alright mb